From 7fc64a1c7cb3100ad9661aa19b4655978538f3ee Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 16:58:40 +0800
Subject: [PORT FROM MAIN] libstagefright: Ignore codec private data to make more clips playable

BZ: 98517

In some MKV files, there is no codec private data for MPEG4 written in
the container. Thus media mime type or video dimension will not be set
in current logic and media framework will treat these clips as malformed
files. Actually, clips can be played fine even without these private data.
To get it fixed, just simply ignore the check of the private data.

Change-Id: I3fc2634b93c8c4eabf5197e0f333ebe602862f7d
Orig-Change-Id: Ie357bf5e1d669b1560c67f36651357379adcffd2
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index e4e92b7..d65938b 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -863,15 +863,13 @@ void MatroskaExtractor::addTracks() {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_AVC);
                     meta->setData(kKeyAVCC, 0, codecPrivate, codecPrivateSize);
                 } else if (!strcmp("V_MPEG4/ISO/ASP", codecID)) {
+                    meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG4);
                     if (codecPrivateSize > 0) {
-                        meta->setCString(
-                                kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG4);
                         addESDSFromCodecPrivate(
                                 meta, false, codecPrivate, codecPrivateSize);
                     } else {
-                        ALOGW("%s is detected, but does not have configuration.",
-                                codecID);
-                        continue;
+                        ALOGW("%s is detected, but does not have configuration.", codecID);
+                        //continue;
                     }
                 } else if (!strcmp("V_VP8", codecID)) {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_VPX);
-- 
1.7.4.1

